### PR TITLE
Polish StandardAnnotationCustomizableTypeExcludeFilters

### PR DESCRIPTION
**Pull request by izeye**
This PR polishes `StandardAnnotationCustomizableTypeExcludeFilter`s by changing to use consistent modifiers.

### CODE REVIEW

File: spring-boot-project/spring-boot-test-autoconfigure/src/main/java/org/springframework/boot/test/autoconfigure/data/ldap/DataLdapTypeExcludeFilter.java [hunk 1]
```diff
@@ -23,9 +23,8 @@
  * {@link TypeExcludeFilter} for {@link DataLdapTest @DataLdapTest}.
  *
  * @author Eddú Meléndez
- * @since 2.2.1
  */
-public final class DataLdapTypeExcludeFilter extends StandardAnnotationCustomizableTypeExcludeFilter<DataLdapTest> {
+class DataLdapTypeExcludeFilter extends StandardAnnotationCustomizableTypeExcludeFilter<DataLdapTest> {
```
Comment (snicoll) [inline spring-boot-project/spring-boot-test-autoconfigure/src/main/java/org/springframework/boot/test/autoconfigure/data/ldap/DataLdapTypeExcludeFilter.java:27]: Ditto

Comment (snicoll) [inline spring-boot-project/spring-boot-test-autoconfigure/src/main/java/org/springframework/boot/test/autoconfigure/data/ldap/DataLdapTypeExcludeFilter.java:27]: Ditto

File: spring-boot-project/spring-boot-test-autoconfigure/src/main/java/org/springframework/boot/test/autoconfigure/data/mongo/DataMongoTypeExcludeFilter.java [hunk 2]
```diff
@@ -23,9 +23,8 @@
  * {@link TypeExcludeFilter} for {@link DataMongoTest @DataMongoTest}.
  *
  * @author Michael Simons
- * @since 2.2.1
  */
-public final class DataMongoTypeExcludeFilter extends StandardAnnotationCustomizableTypeExcludeFilter<DataMongoTest> {
+class DataMongoTypeExcludeFilter extends StandardAnnotationCustomizableTypeExcludeFilter<DataMongoTest> {
```
Comment (snicoll) [inline spring-boot-project/spring-boot-test-autoconfigure/src/main/java/org/springframework/boot/test/autoconfigure/data/mongo/DataMongoTypeExcludeFilter.java:27]: Ditto

Comment (snicoll) [inline spring-boot-project/spring-boot-test-autoconfigure/src/main/java/org/springframework/boot/test/autoconfigure/data/mongo/DataMongoTypeExcludeFilter.java:27]: Ditto

**Review by snicoll — CHANGES_REQUESTED** [review]
Unfortunately, we cannot move these classes to package private without a deprecation period. Deprecating them looks weird though so I am tempted to keep things as they are for now. 

Can you update your PR to harmonize the constructor visibility?

**Comment by izeye** [comment]
@snicoll Their constructor is package-private, so I thought it's effectively package-private. Am I missing something?

**Comment by snicoll** [comment]
In principle, yes but the type being public doesn't prevent anyone from doing something like:

```java
@TypeExcludeFilters(DataMongoTypeExcludeFilter.class)
```

**Pull request closed without merging**